### PR TITLE
Reorder CI targets

### LINKED ISSUE #331: Order of CI targets
The priority CI targets (The ones at the top) need to be adjusted for the latest compilers. And we should ensure we do not duplicate targets (see gcc-11).

### LINKED ISSUE #332: Warnings as errors
We should turn this on CI

### PR DESCRIPTION
**Pull request by alandefreitas**
fix #331, fix #332

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -14,65 +14,74 @@ windowsglobalimage="cppalliance/dronevs2019"
 
 def main(ctx):
   return [
+  # Coverage
+  linux_cxx("Coverage", "g++-8", packages="g++-8", buildscript="drone", buildtype="codecov", image=linuxglobalimage, environment={'COMMENT': 'codecov.io', 'LCOV_BRANCH_COVERAGE': '0', 'B2_CXXSTD': '11', 'B2_TOOLSET': 'gcc-8', 'B2_DEFINES': 'BOOST_NO_STRESS_TEST=1', 'DRONE_JOB_UUID': 'da4b9237ba', "CODECOV_TOKEN": {"from_secret": "codecov_token"}}, globalenv=globalenv),
+  linux_cxx("Coverity Scan", "g++", packages="", buildscript="drone", buildtype="coverity", image=linuxglobalimage, environment={'COMMENT': 'Coverity Scan', 'B2_TOOLSET': 'clang', 'DRONE_JOB_UUID': '4d134bc072'}, globalenv=globalenv),
```
Comment (vinniefalco) [inline .drone.star:19]: The hell with Coverity, they want us to pay. Put them at the bottom, commented out.

File: .drone.star [hunk 2]
```diff
@@ -14,65 +14,74 @@ windowsglobalimage="cppalliance/dronevs2019"
 
 def main(ctx):
   return [
+  # Coverage
```
Comment (vinniefalco) [inline .drone.star:17]: Please paste the chart we made, at the top of this function
```
  # Priorities: (no 2a, no betas)
  #
  # coverage
  # latest gcc: 11,14,17,20
  # latest clang: 11,14,17,20
  # oldest gcc: 4.8
  # oldest clang: 3.8
  # valgrind
  # asan
  # tsan
  # ubsan
  # docs
  # arm64
  # s390x
  # (...the rest)
```

Comment (alandefreitas) [reply .drone.star:26]: I interleaved it with the jobs. But I'll put it there.

File: .drone.star [hunk 3]
```diff
@@ -14,65 +14,82 @@ windowsglobalimage="cppalliance/dronevs2019"
...
 def main(ctx):
   return [
+  # Priorities:
+  # 1. Coverage
+  # 2. Latest compilers
+  # 3. Oldest compilers
+  # 4. Sanitizers
+  # 5. Documentation
+  # 6. Other platforms
+  # 7. The rest
```
Comment (vinniefalco) [inline .drone.star:24]: No!!! This is not what we wrote. Again:
```
  # Priorities: (no 2a, no betas)
  #
  # coverage
  # latest gcc: 11,14,17,20
  # latest clang: 11,14,17,20
  # oldest gcc: 4.8
  # oldest clang: 3.8
  # valgrind
  # asan
  # tsan
  # ubsan
  # docs
  # arm64
  # s390x
  # (...the rest)
```

Comment (alandefreitas) [reply .drone.star:24]: Oh... you really want the ugly version... ok

File: .drone.star [hunk 4]
```diff
@@ -14,65 +14,89 @@ windowsglobalimage="cppalliance/dronevs2019"
...
-  linux_cxx("B2_TOOLSET=gcc-12 B2_CXXSTD=17", "g++-12", packages="g++-12", buildscript="drone", buildtype="boost", image="cppalliance/droneubuntu2204:1", environment={'B2_TOOLSET': 'gcc-12', 'B2_CXXFLAGS': '-funsigned-char', 'B2_CXXSTD': '17,20'}, globalenv=globalenv),
+  # Latest compilers
+  linux_cxx("GCC 12: C++17", "g++-12", packages="g++-12", buildscript="drone", buildtype="boost", image="cppalliance/droneubuntu2204:1", environment={'B2_TOOLSET': 'gcc-12', 'B2_CXXFLAGS': '-funsigned-char', 'B2_CXXSTD': '17,20'}, globalenv=globalenv),
+  linux_cxx("Clang 14: C++11, 14, 17, 20", "clang++-14", packages="clang-14 libstdc++-10-dev", llvm_os="jammy", llvm_ver="14", buildscript="drone", buildtype="boost", image="cppalliance/droneubuntu2204:1", environment={'B2_TOOLSET': 'clang-14', 'B2_CXXSTD': '11,14,17,20'}, globalenv=globalenv),
+  windows_cxx("MSVC 14.3: C++17, 20", "", image="cppalliance/dronevs2022", buildtype="boost", buildscript="drone", environment={"B2_TOOLSET": "msvc-14.3", "B2_CXXSTD": "17,20"},globalenv=globalenv),
```
Comment (vinniefalco) [inline .drone.star:39]: It isn't "Latest compilers" it is
```
  # latest gcc: 11,14,17,20
  # latest clang: 11,14,17,20
```

and there's no MSVC, look at the chart (this is why we pasted it)

File: .drone.star [hunk 5]
```diff
@@ -14,65 +14,89 @@ windowsglobalimage="cppalliance/dronevs2019"
...
+  #
+  # coverage
+  # latest gcc: 11,14,17,20
+  # latest clang: 11,14,17,20
+  # oldest gcc: 4.8
+  # oldest clang: 3.8
+  # valgrind
+  # asan
+  # tsan
+  # ubsan
```
Comment (vinniefalco) [inline .drone.star:27]: edit this list to reflect that we moved valgrind as the last sanitizer

File: .drone.star [hunk 6]
```diff
@@ -14,65 +14,89 @@ windowsglobalimage="cppalliance/dronevs2019"
...
+  # docs
```
Comment (vinniefalco) [inline .drone.star:28]: Edit this list to reflect that we put the docs last

Comment (alandefreitas) [reply .drone.star:28]: We didn't. docs is right after valgrind.

File: .drone.star [hunk 7]
```diff
@@ -14,65 +14,91 @@ windowsglobalimage="cppalliance/dronevs2019"
...
+  # oldest gcc: 4.8
+  # oldest clang: 3.8
+  # asan
+  # tsan
+  # ubsan
+  # valgrind
+  # docs
+  # arm64
+  # s390x
+  # (...the rest)
```
Comment (vinniefalco) [inline .drone.star:31]: add the cmake/install targets to this list, and add your name in the copyright

File: .drone.star [hunk 8]
```diff
@@ -14,65 +14,91 @@ windowsglobalimage="cppalliance/dronevs2019"
...
+  # Documentation
+  linux_cxx("Docs", "g++", packages="docbook docbook-xml docbook-xsl xsltproc libsaxonhe-java default-jre-headless flex libfl-dev bison unzip rsync", buildtype="docs", buildscript="drone", image="cppalliance/droneubuntu1804:1", environment={'COMMENT': 'docs', 'DRONE_JOB_UUID': 'b6589fc6ab'}, globalenv=globalenv),
+
+  # CMake tests (https://github.com/boostorg/boost-ci)
+  ## Compiling as part of the boost superproject
+  linux_cxx("CMake Superproject", "g++", packages="", buildscript="drone", buildtype="cmake-superproject", image="cppalliance/droneubuntu1804:1", globalenv=globalenv),
+  ## Installing
+  linux_cxx("CMake Install", "g++", packages="", buildscript="drone", buildtype="cmake1", image="cppalliance/droneubuntu1804:1", environment={'CMAKE_INSTALL_TEST': '1', 'DRONE_JOB_UUID': 'b6589fc6ab'}, globalenv=globalenv),
+
+  # arm64 (unsigned char)
```
Comment (vinniefalco) [inline .drone.star:61]: Put a big horizontal line in a comment just before this line

File: .drone.star [hunk 9]
```diff
@@ -14,65 +15,95 @@ windowsglobalimage="cppalliance/dronevs2019"
...
+  # latest clang: 11,14,17,20
+  # oldest gcc: 4.8
+  # oldest clang: 3.8
+  # asan
+  # tsan
+  # ubsan
+  # valgrind
+  # docs
+  # cmake superproject
+  # cmake install
```
Comment (vinniefalco) [inline .drone.star:31]: cmake and docs should be at the end of this list, under arm/s390x (and update the drone lines)

Comment (alandefreitas) [reply .drone.star:33]: Changed it

(Even though docs fail orders of magnitude more often than arm/s390x and cmake fail orders of magnitude more often than docs)

**Pull request closed without merging**